crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -448,6 +473,55 @@ fn gcroots_dir(flox: &Flox, pointer: &ManagedPointer) -> PathBuf {
     flox.cache_dir.join("run").join(pointer.owner.to_string())
 }
 
+impl ManagedEnvironment {
+    #[allow(unused)]
+    pub fn pull(&mut self) -> Result<(), ManagedEnvironmentError> {
+        // Fetch the remote branch into FETCH_HEAD
+        self.floxmeta
+            .git
+            .fetch_ref(
+                "origin",
+                &format!("{}:", remote_branch_name(&self.system, &self.pointer)),
+            )
+            .unwrap();
+
+        // Check whether we can fast-forward merge the remote branch into the local branch
+        // In not the environment has diverged.
+        let consistent_history = self
+            .floxmeta
+            .git
+            .branch_contains_commit(
+                &remote_branch_name(&self.system, &self.pointer),
+                "FETCH_HEAD",
+            )
+            .map_err(ManagedEnvironmentError::Git)?;
+        if !consistent_history {
+            Err(ManagedEnvironmentError::Diverged)?;
+        }
+
+        // try fast forward merge local env branch into project branch
+        self.floxmeta
+            .git
+            .push2(
+                ".",
+                format!(
+                    "FETCH_HEAD:refs/heads/{project_branch}",
+                    project_branch = branch_name(&self.system, &self.pointer, &self.path)?
+                ),
+            )
+            .unwrap();
+
+        lock_env(
```

Comment:
suggestion blocking: add a TODO to not do another fetch or change `lock_env` to `write_lockfile` or something that doesn't run another fetch